src/wasm/wasm-emscripten.cpp

```diff
@@ -584,10 +584,14 @@ struct FixInvokeFunctionNamesWalker : public PostWalker<FixInvokeFunctionNamesWa
       return;
 
     FunctionType* func = wasm.getFunctionType(curr->functionType);
-    Name newname = fixEmEHSjLjNames(curr->name, getSig(func));
+    Name newname = fixEmEHSjLjNames(curr->base, getSig(func));
     if (newname == curr->name)
       return;
 
+    if (curr->base != curr->name) {
+      Fatal() << "Import name and function name to not match: '" << curr->base << "' '" << curr->name << "'";
```

Comment:
nit: **d**o not match

Also should this check be before the call to fixEmEHSjLjNames?

Reply:
I was thinking that we only really care when there are functions that we want to rename.